en/computer_vision/obstacle_avoidance.md

```diff
@@ -0,0 +1,156 @@
+
+The desired route comes from a [ROS](http://dev.px4.io/en/ros/) node running on a companion computer.
+This is passed into an obstacle avoidance module (another ROS node).
+The avoidance software sends the planned path to the flight stack as a stream of `SET_POSITION_TARGET_LOCAL_NED` messages.
+
+> **Note** The only required PX4-side setup is to put PX4 into *Offboard mode*..
+  While the `SET_POSITION_TARGET_LOCAL_NED` setpoints come from a ROS collision avoidance node, to PX4 could be from any MAVLink system.
+
+The tested hardware/software platform is [Intel Aero](../flight_controller/intel_aero.md) running either the *local_planner* or *global_planner*. 
+The setup is as described in the [Intel Aero > Obstacle Avoidance](../flight_controller/intel_aero.md#obstacle-avoidance) and in the [PX4/avoidance](https://github.com/PX4/avoidance#obstacle-detection-and-avoidance) Github repo.
+
+
+## Mission Mode Avoidance {#mission_mode}
+
+PX4 supports obstacle avoidance in [Missions](../flight_modes/mission.md), using avoidance software running on a separate companion computer. 
+
+Obstacle avoidance is enabled within PX4 by [setting](../advanced_config/parameters.md) the `MPC_OBS_AVOID` to 1.
+PX4 communicates with the obstacle avoidance software using an implementation of the MAVLink [Path Planning Protocol](TBD) (Trajectory Interface) which is [#described below](mission_avoidance_interface).
+Provided an avoidance system complies with this interface it can be used with PX4.
+
+The tested companion computer platform is [Intel Aero](../flight_controller/intel_aero.md) running either the *local_planner* or *global_planner* avoidance software.
+This is set up as described in the [Intel Aero > Obstacle Avoidance](../flight_controller/intel_aero.md#obstacle-avoidance) and in the [PX4/avoidance](https://github.com/PX4/avoidance#obstacle-detection-and-avoidance) Github repo.
+
+
+### Mission Progression
+
+The companion computer provides setpoint navigation for the *entire route* (not just when there are obstacles) based on current position, current waypoint and next waypoint information from PX4.
+
+The resulting mission behaviour is very similar to missions without obstacle avoidance, and mainly affects the criteria used to determine that a waypoint has been reached.
+
+The difference when avoidance is active are:
+- A waypoint is "reached" when the vehicle is within the acceptance radius, regardless of its heading.
+  - In normal missions the vehicle must reach a waypoint with a certain heading (i.e. in a "close to" straight line from the previous waypoint). 
+  - When obstacle avoidance is active, this constraint cannot be fulfilled because the obstacle avoidance algorithm has full control of the vehicle heading, and the vehicle always moves in the current field of view. 
+- navigator updates the triplets when the vehicle has reached the acceptance radius of each waypoint. 
```

Comment:
What triplets?

Reply:
navigator works with triplets (previous, current, next) waypoint. The current waypoint is the avoidance goal.